project_euler/problem_112/sol1.py

```diff
@@ -0,0 +1,65 @@
+"""
+Problem 112: https://projecteuler.net/problem=112
+
+Working from left-to-right if no digit is exceeded by the digit to its left it is
+
+called an increasing number; for example, 134468.
+Similarly if no digit is exceeded by the digit to its right it is called a decreasing
+number; for example, 66420.
+We shall call a positive integer that is neither increasing nor decreasing a "bouncy"
+number, for example, 155349.
+Clearly there cannot be any bouncy numbers below one-hundred, but just over half of
+the numbers below one-thousand (525) are bouncy. In fact, the least number for which
+the proportion of bouncy numbers first reaches 50% is 538.
+Surprisingly, bouncy numbers become more and more common and by the time we reach
+21780 the proportion of bouncy numbers is equal to 90%.
+
+Find the least number for which the proportion of bouncy numbers is exactly 99%.
+"""
+
+
+def check_bouncy(n: int) -> bool:
+    """
+    Returns True if number is bouncy, False otherwise
+    >>> check_bouncy(6789)
+    False
+    >>> check_bouncy(-12345)
+    False
+    >>> check_bouncy(6.74)
+    False
+    >>> check_bouncy(132475)
+    True
+    >>> check_bouncy(-6548)
+    True
+    """
+    if not isinstance(n, int):
+        return False
+    return "".join(sorted(str(n))) != str(n) and "".join(sorted(str(n)))[::-1] != str(n)
+
+
+def solution(percent: int = 99) -> int:
+    """
+    Returns the least number for which the proportion of bouncy numbers is
+    exactly 'percent'
+    >>> solution(50)
+    538
+    >>> solution(90)
+    21780
+    >>> solution(80)
+    4770

```

Comment:
What happens when the `percent` variable is below 50?

Reply:
Or is it ok if the percent variable goes above 100? 
It depends on what it is representing, please figure it out.

Reply:
Actually, if you pass an input which is greater than 99, it will take a very long time to produce any output. 
And if you passed **100**, it might never converge to any output (probably the computer will crash)

Reply:
Then it should probably be restricted and if it goes beyond the range, it should raise `ValueError` with an appropriate message otherwise a person calling this function with an input greater than 100 will be stuck forever thinking "Is this function doing anything?" 😂

Reply:
OK!